### PR TITLE
[V2] Query parser: add split function

### PR DESCRIPTION
**Pull request by shifucun**

### CODE REVIEW

File: internal/server/v2/parser.go [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package v2
+package v2
+
+import "fmt"
+
+// split a string by "->" and "<-"
+func split(s string) ([]string, error) {
```
Comment (Spaceenter) [inline internal/server/v2/parser.go:21]: I think we also need to split by "," and/or ":" etc., when parsing other parts of the query. So maybe make the function name more specific?

File: internal/server/v2/parser.go [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+	if len(s) < 2 {
+		return nil, fmt.Errorf("invalid query string: %s", s)
```
Comment (Spaceenter) [inline internal/server/v2/parser.go:23]: Wonder if we want to use more formal error syntax. For example:

status.Errorf(codes.InvalidArgument, ".....")

https://pkg.go.dev/google.golang.org/grpc/codes

**Review by Spaceenter — APPROVED** [review]

File: internal/server/v2/parser.go (hunk 1 above)
Comment (shifucun) [reply internal/server/v2/parser.go]: changed to splitArc

File: internal/server/v2/parser.go (hunk 2 above)
Comment (shifucun) [reply internal/server/v2/parser.go]: Done

**Pull request merged**